Update Contributing guidelines (#1417)
nodejs/nodejs.dev: CONTRIBUTING.md

* [`squash`][] pull-requests made up of multiple commits
* Land how you like as long as there are no merge commits

## How to become a collaborator

* Collaborators must be actively contributing to the project

> * A collaborator PR must be opened on the @nodejs/nodejs.dev with user's github name in the title

Reviewer: ```suggestion
* A PR must be opened on the @nodejs/nodejs.dev README file adding the new collaborator to the list (note the order of names)
```

> * A collaborator issue must be opened on the @nodejs/nodejs.dev with user's github name in the title

Author: ok

> * Issue must be reviewed and approved by at least two members of @nodejs/nodejs.dev, @nodejs/website, or @nodejs/website-redesign 

Reviewer: ```suggestion
* The Pull Request must be approved by at least two members of @nodejs/nodejs.dev, @nodejs/website, or @nodejs/website-redesign 
```

> * PR must remain open for 48 hours without any objections

Reviewer: ```suggestion
* PR must remain open for 72 hours without any objections
```

> * After the issue is reviewed and approved it must stay for 48 hours without any objections

Author: i dont know where its written but i've heard that in conversation with people

> * Once the requirements are met and approved, a member of @nodejs/community-committee will make the person an official collaborator

Reviewer: What does "make the person an official collaborator" mean? If this request is done through a PR to add the person to the list, that is step one. I believe the only other step is to add them to the team that has access to this PR.

Author: it's done through the PR

Reviewer: If its done through the PR, then what does this step mean? I think what is perhaps missing is adding them to the github team that manages this repo.

Reviewer: ```suggestion
* Once the requirements are met and approved, the pull request can be merged and a member of @nodejs/community-committee will add the new member to the @nodejs/nodejs-dev team
```
